Add NLS To ICU Transition Scenarios (#36832)
dotnet/docs: docs/core/extensions/globalization-icu.md

## ICU on Windows

Windows now incorporates a pre-installed [icu.dll](/windows/win32/intl/international-components-for-unicode--icu-) version as part of its features which is automatically employed for globalization tasks. This modification allows .NET to leverage this ICU library for its globalization support. In cases where the ICU library is unavailable or cannot be loaded, as is the case with older Windows versions, .NET 5 and subsequent versions revert to using the NLS-based implementation.

> The following table furnishes details regarding which versions of .NET are capable of loading the ICU library across different Windows client and server versions:

Reviewer: ```suggestion
The following table shows which versions of .NET are capable of loading the ICU library across different Windows client and server versions:
```

> .NET Version| Windows Version

Reviewer: ```suggestion
.NET version| Windows version
```

---|---

> .NET 5.0 or .NET 6.0|Windows client 10 version 1903 or later

Reviewer: ```suggestion
.NET 5 or .NET 6|Windows client 10 version 1903 or later
```

> .NET 5.0 or .NET 6.0|Windows Server 2022 or later

Reviewer: ```suggestion
.NET 5 or .NET 6|Windows Server 2022 or later
```
